### DISCUSSION TITLE
Add 'sampleWith' operator

### DISCUSSION THREAD

**Discussion by j-mueller** [ReactiveX/rxjs#6681] [Ideas / Feature request]
The `sample` function currently drops the value of the notifier. It would be nice to be able to keep the notifier value like this:

```typescript
sampleWith<T, A>(notifier: Observable<A>): OperatorFunction<T, Cons<T, A>>
```

We could then write the old `sample` (ignoring the notifier value) in terms of `sampleWith`.

**Comment by voliva** [comment]
Wouldn't `withLatestFrom` help?

`source |> sampleWith(notifier)` is almost the same as `notifier |> withLatestFrom(source)`.

The only difference is that two emissions of notifier will result in two emissions even if source didn't change...

But then, wouldn't `source |> sample(notifier) |> withLatestFrom(notifier)` do the job?
